### PR TITLE
weatherman-finalized

### PR DESCRIPTION
**Pull request by zaini63033**

### CODE REVIEW

File: weather-2.mjs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            const values = lines[i].split(',');
+            const obj = new Map();
+            for(let j = 0; j < values.length; j++) {
+                obj.set(headers[j], values[j]);
+            }
+            arr.push(obj);
+        }
+
+        for(let i = 0; i < arr.length; i++) {
+            const currentDate = parseInt(arr[i].get(headers[0]));
```
Comment (salmanR107) [inline weather-2.mjs]: This isn't being used, remove it.

File: weather-2.mjs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            if(!isNaN(hHumid)) humid += hHumid;
+        }
+
+        tempH /= arr.length;
+        tempL /= arr.length;
+        humid /= arr.length;
+
+        console.log("Highest mean humidity: " + humid.toString() + "%");
+        console.log("Highest mean temperature: " + tempH.toString() + "°C");
+        console.log("Lowest mean temperature: " + tempL.toString() + "°C");
```
Comment (salmanR107) [inline weather-2.mjs]: Use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead. 
Ex: `Highest mean humidity: ${humid}%`

File: weather-3.mjs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            let lstring = '';
+
+            for(let j = 0; !isNaN(parseInt(hTemp)) && j < parseInt(hTemp); j++)
+                hstring += '+';
+
+            for(let j = 0; !isNaN(parseInt(lTemp)) && j < parseInt(lTemp); j++)
+                lstring += '*';
+
+            console.log(currentDate + ' ' + hstring + ' ' + hTemp);
+            console.log(currentDate + ' ' + lstring + ' ' + lTemp);
```
Comment (salmanR107) [inline weather-3.mjs]: Use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead.

File: weather-4.mjs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            let hstring = '';
+            let lstring = '';
+
+            for(let j = 0; !isNaN(parseInt(hTemp)) && j < parseInt(hTemp); j++)
+                hstring += '+';
+
+            for(let j = 0; !isNaN(parseInt(lTemp)) && j < parseInt(lTemp); j++)
+                lstring += '*';
+
+            console.log(currentDate + ' ' + hstring + lstring + ' ' + lstring + lTemp + '-' + hTemp);
```
Comment (salmanR107) [inline weather-4.mjs]: Use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead.

**Review by salmanR107 — CHANGES_REQUESTED** [review]

File: weather-2.mjs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        tempH /= arr.length;
+        tempL /= arr.length;
+        humid /= arr.length;
+
+        console.log(`
+            Highest mean humidity: ${humid}"%
+            Highest mean temperature: ${tempH}°C
+            Lowest mean temperature: ${tempL}"°C`);
```
Comment (salmanR107) [inline weather-2.mjs]: Do we need the double quotation marks **"** before % and C?

File: weather-3.mjs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            }
+            arr.push(obj);
+        }
+
+        for(let i = 0; i < arr.length; i++) {
+            const currentDate = arr[i].get(headers[0]);
+            const hTemp = arr[i].get(headers[1]);
+            const lTemp = arr[i].get(headers[3]);
+
+            // Check for empty strings
```
Comment (salmanR107) [inline weather-3.mjs]: Please remove

File: weather-2.mjs (hunk 5 above)
Comment (zaini63033) [reply weather-2.mjs]: nope, prolly a typo. Lemme fix it

**Review by salmanR107 — APPROVED** [review]

**Pull request merged**